### ISSUE TITLE
Error when sending sales mails in async mode if product was removed

### ISSUE THREAD

**Issue by ilnytskyi** [magento/magento2#26876]
### Preconditions (*)

1. magento 2.3.1+
2. 

### Steps to reproduce (*)

1. Set up async mails sending 
2. Buy a product, create order, invoice or shipment
3. Send sales mails
4. Remove product (along with configurable(parent) product)
5. Try to resend sales email (e.g. set column `email_sent` to Null in relative tables)
6. or try to send invoice/shipment again 
![Selection_375](https://user-images.githubusercontent.com/3192770/74532868-92509300-4f30-11ea-8607-13153fe5712f.png)

### Expected result (*)

1. Emails are send
2. If there are newer sales items planned to send (email_sent is null) emails will be sent
3. Job code shows warning or problem item is ignored but newer items are sent

### Actual result (*)

1. Emails are not sent
2. If there are newer sales items planned to send (email_sent is null) emails will not be sent; It leads that sales async mailing stop working
![Selection_374](https://user-images.githubusercontent.com/3192770/74533150-32a6b780-4f31-11ea-8237-d6100f07e81c.png)
3. in `var/log/cron.log` this log appears 
```
[2020-02-14 12:41:09] main.ERROR: Cron Job sales_send_order_invoice_emails has an error: Call to a member function getData() on null. Statistics: {"sum":0,"count":1,"realmem":0,"emalloc":0,"realmem_start":73400320,"emalloc_start":36940360} [] []
[2020-02-14 12:41:09] main.CRITICAL: Error when running a cron job {"exception":"[object] (RuntimeException(code: 0): Error when running a cron job at /var/www/html/vendor/magento/module-cron/Observer/ProcessCronQueueObserver.php:327, Error(code: 0): Call to a member function getData() on null at /var/www/html/vendor/magento/module-catalog/Helper/Image.php:502)"} []

```
4. Newer items are blocked to send

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sudheers-kensium 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-31919` were created

**Issue Available**: @sudheers-kensium, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @ilnytskyi. Thank you for your report. 
The issue has been fixed in magento/magento2#26877 by @ilnytskyi in 2.4-develop branch
Related commit(s):
 - [405dfec0026e6be2c11281d31774f08a293ded2d](https://github.com/magento/magento2/commit/405dfec0026e6be2c11281d31774f08a293ded2d)
 - [f8a5126f32f5c6e1eeedeff893873974ae8cb6d4](https://github.com/magento/magento2/commit/f8a5126f32f5c6e1eeedeff893873974ae8cb6d4)
 - [da381030bdeca058a6979142fa1ac265628f9f1d](https://github.com/magento/magento2/commit/da381030bdeca058a6979142fa1ac265628f9f1d)

The fix will be available with the upcoming 2.4.2 release.
